### PR TITLE
Added Caching of npm packages and added npm install

### PR DESCRIPTION
**Pull request by ohbus**
- [x] Added `npm install` for installing node packages
- [x] Added naming for different actions
- [x] Added caching support of `node_modules` to not download packages for every build.

> Let me know what needs to be changed. Thanks

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
 
-name: Node.js CI
+name: Sclr.ac CI
```
Comment (championswimmer) [inline .github/workflows/node.js.yml]: Node.js CI was a better name it defines what this action is, not what this project is 😅

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -12,18 +12,40 @@ on:
...
   build:
 
-    runs-on: ubuntu-latest #Can change the runtime environment according to needs
+    runs-on: ${{ matrix.os }}
 
     strategy:
       matrix:
         node-version: [14.x, 16.x]
         # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+        os: [ubuntu-latest, macOS-latest, windows-latest]
```
Comment (championswimmer) [inline .github/workflows/node.js.yml]: this will eat into our free github action minutes. just ubuntu is fine.

Comment (ohbus) [reply .github/workflows/node.js.yml]: As you mentioned it is free, so why not use it 😄

Comment (championswimmer) [reply .github/workflows/node.js.yml]: Only 2000 minutes are free. Not unlimited :( 

Also we should not compute just for fun. There is a carbon footprint and environment impact of such small steps. Do we have any potential for cross-OS issues right now? If not why waste computation resources.

**Review by championswimmer — CHANGES_REQUESTED** [review]

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -12,18 +12,40 @@ on:
...
+    
     - name: Use Node.js ${{ matrix.node-version }}
       uses: actions/setup-node@v2
       with:
         node-version: ${{ matrix.node-version }}
-    - run: npm ci
-    - run: npm run build
+    
+    - name: Install Node packages
+      run: npm install
```
Comment (championswimmer) [inline .github/workflows/node.js.yml]: npm ci should run npm install right ?

**Review by championswimmer — CHANGES_REQUESTED** [review]

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (ohbus) [reply .github/workflows/node.js.yml]: I thought 2000 minutes are free for private repo.

But I agree with your point on carbon footprint but unless we see any raised issue about the same let us stick with ubuntu

File: .github/workflows/node.js.yml (hunk 3 above)
Comment (ohbus) [reply .github/workflows/node.js.yml]: I wouldn't know, I used to follow `npm install` to install the node_modules.

Please tell me what to do?

Comment (championswimmer) [reply .github/workflows/node.js.yml]: Use NPM CI 

It is used for installing on CI machines 
https://www.geeksforgeeks.org/difference-between-npm-i-and-npm-ci-in-node-js/

Comment (ohbus) [reply .github/workflows/node.js.yml]: Thanks for the information

**Pull request merged**